applications/GeoMechanicsApplication/custom_constitutive/thermal_law.h

```diff
@@ -0,0 +1,61 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//                   Gennady Markelov
+//
+
+#pragma once
+
+#include "geo_mechanics_application_variables.h"
+#include "includes/serializer.h"
+
+namespace Kratos
+{
+
+/**
+ * @class GeoThermalDispersionLaw
+ * @ingroup GeoMechanicsApplication
+ * @brief This class defines the thermal dispersion for heat cases
+ */
+class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoThermalLaw
+{
+public:
+    /// Counted pointer of GeoThermalLaw
+    KRATOS_CLASS_POINTER_DEFINITION(GeoThermalLaw);
+
+    GeoThermalLaw() {}
+
+    virtual ~GeoThermalLaw() = default;
+
+    virtual GeoThermalLaw::Pointer Clone() const = 0;
```

Comment:
I don't think we're still using the Clone and WorkingSpaceDimension functions anywhere. If that's indeed the case, I'd suggest to remove the functions